### DISCUSSION TITLE
Custom columns in Users table not usable in REST API queries

### DISCUSSION THREAD

**Discussion by eftomi** [Budibase/budibase#8824] [Help]
Hi, if you add custom column in Users table, the value from it is not "visible" to headers and parameters of a REST API query. 

For instance, if I have a custom text column "AdditionalInfo" and use it as a parameter AdditionalInfo in a query as 

`{{ Current User.AdditionalInfo }}`

the receiving end gets an empty string:

```
"query": {
  "AdditionalInfo": ""
}
```

I tried it out with different tests like running the query by a button on a screen, app preview as admin, as yourself, with published app. Other elements on the same screen (e.g. paragraphs) get proper data from cutom fields without any problems. I'm using self hosted Budibase version 2.1.31.

**Comment by andz-bb** [comment]
hey @eftomi, 

I was able to replicate what you were seeing on my end but I have found a workaround. If you define a binding in your REST query, say `AdditionalInfoBinding`, and provide a default for it, you can then use that binding as a parameter of the query.

<img width="997" alt="Screenshot 2022-11-28 at 13 34 09" src="https://user-images.githubusercontent.com/110921612/204290647-d7d0400f-74f3-427e-ae6c-0cc7359ad261.png">
<img width="1006" alt="Screenshot 2022-11-28 at 13 33 35" src="https://user-images.githubusercontent.com/110921612/204290676-f05477ee-9895-4fb4-8043-66cab9e99ff4.png">


Then where you are executing the query, say from a button action, you can provide the AdditionalInfo column from the User table as the value for the query binding.

<img width="1169" alt="Screenshot 2022-11-28 at 13 38 33" src="https://user-images.githubusercontent.com/110921612/204291514-20aea76b-8d1e-49a1-8304-c1ab7f5f5040.png">

In my case the default value for my query binding is 4, but the value from my user is 5. With this binding set up, it works as expected:

<img width="581" alt="Screenshot 2022-11-28 at 13 45 48" src="https://user-images.githubusercontent.com/110921612/204292947-b359be94-ea34-4f5e-be7d-2d86f545c3aa.png">

Hope this makes sense OK, if you have any further questions on this let me know and I'll be happy to help 😊

**Comment by eftomi** [comment]
Thanks Andrew for your detailed suggestion. This works in general, however I'm using the query in question to "chain data" between queries, it provides values for dynamic variables. If this query gets re-evaluated, then it uses the default value which isn't OK. 

Another option would be to use a state variable, but these cannot be used in REST API as far as I know.

**Comment by evtnl** [comment]
see also [7368](https://github.com/Budibase/budibase/issues/7368) It seems this is by design. You can only add custom user fields, but not use them like other fields
